### PR TITLE
Add option to the interactions panel to trigger the belch emote on prey resisting.

### PR DESCRIPTION
**Pull request by NickBelmont**
## About The Pull Request
Adds a new option to the interactions tab of the vorepanel to allow for setting a % chance for the belch emote to occur when prey resist.
## Changelog
:cl:
add: Belch chance parameter to the interactions tab of the vorepanel
add: Prey resisting can trigger a belch emote if interactions are set up to allow it.
/:cl:

### CODE REVIEW

File: code/modules/vore/eating/vorepanel_vr.dm [hunk 1]
```diff
@@ -3603,6 +3608,11 @@ var/global/list/belly_colorable_only_fullscreens = list("a_synth_flesh_mono",
 			if(!isnull(escape_chance_input)) //These have to be 'null' because both cancel and 0 are valid, separate options
 				host.vore_selected.escapechance = sanitize_integer(escape_chance_input, 0, 100, initial(host.vore_selected.escapechance))
 			. = TRUE
+		if("b_belchchance")
+			var/belch_chance_input = tgui_input_number(user, "Set chance for belch emote on prey resist (as %)", "Resist Belch Chance", null, 100, 0)
+			if(!isnull(belch_chance_input))
```
Comment (Kashargul) [inline code/modules/vore/eating/vorepanel_vr.dm:3613]: should be given the previous value as default

**Comment by Kashargul** [comment]
The addition is missing in the im/export. E.g. Exportpanel UI

**Comment by NickBelmont** [comment]
I believe that should fix most of the issues. Could have still missed something relating to importing or exporting options though, so let me know if that's the case. The input also now defaults to the current value.

**Comment by NickBelmont** [comment]
Also realized I forgot to remove the changes I accidentally made to the (seemingly?) unused version of the vorepanel UI that's not in the chompstation folder. Should those be reverted?

**Comment by Kashargul** [comment]
> Also realized I forgot to remove the changes I accidentally made to the (seemingly?) unused version of the vorepanel UI that's not in the chompstation folder. Should those be reverted?

So not touch the non chompstation vorepanel at all, in both cases changes are only allowed in the modular file (chompstation UI folder).

**Comment by NickBelmont** [comment]
Right, that's what I thought, just wanted to make sure. Reverted any changes that I made to those.

**Comment by Kashargul** [comment]
> Right, that's what I thought, just wanted to make sure. Reverted any changes that I made to those.

👍, gonna review it once I have the time

File: code/modules/vore/eating/belly_obj_vr.dm [hunk 2]
```diff
@@ -1991,7 +1993,10 @@
 		else
 			playsound(src, struggle_rustle, vary = 1, vol = 75, falloff = VORE_SOUND_FALLOFF, frequency = noise_freq, preference = /datum/preference/toggle/digestion_noises, volume_channel = VOLUME_CHANNEL_VORE) //CHOMPEdit
 	//CHOMPEdit End
-
+	//CHOMPEdit Start
```
Comment (Kashargul) [inline code/modules/vore/eating/belly_obj_vr.dm:1996]: End and start following each other, remove those two comments and keep the section going

**Review by Kashargul — CHANGES_REQUESTED** [review]

File: code/modules/vore/eating/belly_obj_vr.dm (hunk 2 above)
Comment (NickBelmont) [reply code/modules/vore/eating/belly_obj_vr.dm]: Got it, should be fixed now.

**Comment by NickBelmont** [comment]
Apologies about the slight mess with the commits, I can squash some of them and update the PR once any remaining issues are cleared up.

**Comment by Kashargul** [comment]
> Apologies about the slight mess with the commits, I can squash some of them and update the PR once any remaining issues are cleared up.

No need to, we squash on merge anyway

File: code/modules/vore/eating/belly_obj_vr.dm [hunk 3]
```diff
@@ -41,6 +41,7 @@
 	var/escape_stun = 0						// AI controlled mobs with a number here will be weakened by the provided var when someone escapes, to prevent endless nom loops
 	var/transferchance = 0 					// % Chance of prey being trasnsfered, goes from 0-100%
 	var/transferchance_secondary = 0 		// % Chance of prey being transfered to transferchance_secondary, also goes 0-100%
+	var/belchchance = 0						//CHOMPAdd % Chance of pred belching on prey struggle
```
Comment (Kashargul) [inline code/modules/vore/eating/belly_obj_vr.dm:44]: personally I would prefer that in belly_obj_ch, the rest looks good, now where I have the time to go through it properly

Comment (NickBelmont) [reply code/modules/vore/eating/belly_obj_vr.dm]: Good to hear, I can see why that would be better placement. I'll push that change through in a moment.

**Comment by NickBelmont** [comment]
I think that all issues I've been made aware of have been resolved with that last commit. If there are any other changes or corrections that need to be made upon further review, just let me know.

**Review by Kashargul — APPROVED** [review]

**Pull request merged**